stable/rails/templates/ingress.yaml

```diff
@@ -1,33 +1,45 @@
 {{ if (and .Values.webserver.enabled .Values.ingress.enabled) }}
+{{- range $index, $list := .Values.ingress.list }}
+---
 apiVersion: extensions/v1beta1
 kind: Ingress
 metadata:
-  name: {{ include "rails.name" . }}-ingress
+  name: {{ include "rails.name" $ }}-ingress-{{ .type }}-{{ $index }}
   annotations:
-  {{- range $key, $value := .Values.ingress.annotations }}
+    {{- if .deny_locations }}
+    nginx.ingress.kubernetes.io/server-snippet: |
+      {{- range .deny_locations }}
+        location {{ . }} {
+          deny all;
+        }
+      {{- end }}
+    {{- end }}
+    kubernetes.io/ingress.class: {{ .type }}
+  {{- range $key, $value := .annotations }}
```

Comment:
For more robust implementation we should:
- skip `server-snippet` annotation from here
- merge `server-snippet` annotation with one defined above if available